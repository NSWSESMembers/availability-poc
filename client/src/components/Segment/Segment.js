import React from 'react';
import SegmentedControlTab from 'react-native-segmented-control-tab';
import PropTypes from 'prop-types';

const Segment = ({ values = [], selectedIndex, handleIndexChange }) => (
  <SegmentedControlTab
    values={values}
    selectedIndex={selectedIndex}
    onTabPress={handleIndexChange}
    tabStyle={{ backgroundColor: 'white', borderColor: '#1e3c78' }}
    activeTabStyle={{ backgroundColor: '#1e3c78' }}
    tabTextStyle={{ color: '#1e3c78' }}
  />
);

Segment.propTypes = {
  values: PropTypes.arrayOf(PropTypes.string),
  selectedIndex: PropTypes.number,
  handleIndexChange: PropTypes.func,
};

export default Segment;
