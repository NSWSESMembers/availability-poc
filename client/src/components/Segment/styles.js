import { extendAppStyleSheet } from '../../screens/style-sheet';

const styles = extendAppStyleSheet({});

export default styles;
