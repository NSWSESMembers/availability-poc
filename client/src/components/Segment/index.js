import Segment from './Segment';
import styles from './styles';

export { Segment, styles };
