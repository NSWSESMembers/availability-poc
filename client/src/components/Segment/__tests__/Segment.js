import React from 'react';
import renderer from 'react-test-renderer';

import Segment from '../Segment';

const noop = () => {};

test('renders correctly', () => {
  const tree = renderer.create(
    <Segment
      values={['one', 'two', 'three']}
      selectedIndex={1}
      handleIndexChange={noop}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
