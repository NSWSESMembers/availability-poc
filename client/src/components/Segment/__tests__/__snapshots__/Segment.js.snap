// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  removeClippedSubviews={false}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "flexDirection": "row",
      },
      Object {},
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomLeftRadius": 5,
        "borderColor": "#1e3c78",
        "borderRightWidth": 0,
        "borderTopLeftRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#0076FF",
            },
            Object {
              "color": "#1e3c78",
            },
            Object {},
          ]
        }
      >
        one
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1e3c78",
        "borderColor": "#1e3c78",
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": -1,
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#0076FF",
            },
            Object {
              "color": "#1e3c78",
            },
            Array [
              Object {
                "color": "white",
              },
              Object {},
            ],
          ]
        }
      >
        two
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderBottomRightRadius": 5,
        "borderColor": "#1e3c78",
        "borderLeftWidth": 0,
        "borderTopRightRadius": 5,
        "borderWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
        "paddingVertical": 5,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#0076FF",
            },
            Object {
              "color": "#1e3c78",
            },
            Object {},
          ]
        }
      >
        three
      </Text>
    </View>
  </View>
</View>
`;
